internal/exec/resolvable/resolvable.go

```diff
@@ -228,13 +228,17 @@ func (b *execBuilder) makeObjectExec(typeName string, fields schema.FieldList, p
 
 	Fields := make(map[string]*Field)
 	rt := unwrapPtr(resolverType)
+	fieldsCount := getFieldCount(rt, map[string]int{})
 	for _, f := range fields {
-		fieldIndex := -1
+		var fieldIndex []int
 		methodIndex := findMethod(resolverType, f.Name)
 		if b.schema.UseFieldResolvers && methodIndex == -1 {
-			fieldIndex = findField(rt, f.Name)
+			if fieldsCount[strings.ToLower(stripUnderscore(f.Name))] > 1 {
+				return nil, fmt.Errorf("%s does not resolve %q: ambiguous field %q", resolverType, typeName, f.Name)
+			}
```

Comment:
Is there a unit test about this error?

Reply:
Yes, added [here](https://github.com/graph-gophers/graphql-go/pull/338/files#diff-d7906d668830f592334a2a80cf6d298cR3030).